tests/integration_tests/functional/test_rng.py

```diff
@@ -10,7 +10,14 @@
 from framework.utils import check_entropy, get_kernel_version
 from framework.utils_cpuid import get_instance_type
 
-INSTANCE_TYPE = get_instance_type()
+# get_instance_type only works on EC2 machines, as it uses IMDSv2.
+# If the call fails, set the constant to `None`, which will cause
+# all tests to run.
+try:
+    INSTANCE_TYPE = get_instance_type()
+except:
+    INSTANCE_TYPE = None
+
 HOST_KERNEL = get_kernel_version(level=1)
```

Comment:
I think this makes more sense to be fixed inside `get_instance_type` itself, since I'm pretty sure other things use this function or will use it in the future.

Reply:
You're right. @pb8o pointed out to me that the correct solution to this already exists, so I switched to some code he sent me :)